pynest/examples/spatial/grid_iaf_irr.py

```diff
@@ -31,30 +31,27 @@
 import nest
 import pylab
 import random
-import nest.topology as topo
 
 nest.ResetKernel()
 
-# generate list of 12 (x,y) pairs
-pos = [[random.uniform(-0.75, 0.75), random.uniform(-0.5, 0.5)]
-       for j in range(12)]
+pos = nest.spatial.free([[random.uniform(-0.75, 0.75), random.uniform(-0.5, 0.5)]
```

Comment:
maybe use nest.random.uniform here?

Reply:
numpy.random.uniform?

Reply:
@Silmathoron with the new NEST-3 we have made random distributions from NEST accessible from PyNEST, and seeing as this is an example I think we should use the NEST version.

Reply:
The problem here is that we want different distributions for different dimensions, which currently can't be done when passing `nest.random` functions to `nest.spatial.free()`. But it shouldn't be difficult to add this, maybe after this PR is merged, then we can update this example.

Reply:
Yesterday, I asked @hakonsbm about this and he recommended to use numpy.random.uniform in this case because with nest.random.uniform it is apparently not possible, yet, to define different min and max for the two dimensions. If there is a better solution, please let me know and I will adjust it.

Reply:
Yes, passing a `nest.random` function to `nest.spatial.free()` will currently use the same distribution in all dimensions. But it shouldn't be difficult to add the possibility to have different distributions for different dimensions. Maybe after this PR is merged, then we can update this example?

Reply:
@jhnnsnk @hakonsbm Ah, I missed that. I think we should be able to have different distributions for different dimensions, but I agree that we can do it after this PR is merged. I will add a note on the project page, and merge this PR into the NEST-3 branch.